### PR TITLE
refactor: migrate RadioButton, TextArea components to TS

### PR DESCRIPTION
**Pull request by PavelKirvelTextkernel**
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. Use imperative, present tense in your commit description ("change", not "changed" or "changes") without uppercases or period (.) at the end.

# Checklist
- [ ] The implementation has been manually tested and complies with Textkernel [browser support guidelines](https://textkernel.com/browser-support/)
- [ ] The implementation complies with [accessibility](CONTIRBUTING.md#accessibility) standards.
- [ ] The component has a [displayName](CONTRIBUTING.md#display-names) defined.
- [ ] The component comes with a detailed [PropTypes](CONTRIBUTING.md#component-props) (and defaultProps) definition.
- [ ] Component PropTypes are sufficiently described / documented.
- [ ] There is [a story](CONTRIBUTING.md#component-showcases) in Storybook.

### CODE REVIEW

File: src/components/Checkbox/__tests__/Checkbox.spec.tsx [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import * as React from 'react';
 import toJson from 'enzyme-to-json';
+import { mount } from 'enzyme';
```
[the remark this answers is not in the document]
Comment (PavelKirvelTextkernel) [reply src/components/Checkbox/__tests__/Checkbox.spec.tsx:3]: I think for .ts files we need to use something like this: https://github.com/eslint/eslint/issues/10803

File: src/components/RadioButton/RadioButtonGroup.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import { bem } from '../../utils';
+import styles from './RadioButtonGroup.scss';
+
+export interface Props {
+    /** The name of the group this radio button belongs to */
+    name: string;
+    /** common onChange handler that will be passed to all children */
+    onChange?: () => void;
+    /** The radio buttons */
+    children: any;
```
[the remark this answers is not in the document]
Comment (PavelKirvelTextkernel) [reply src/components/RadioButton/RadioButtonGroup.tsx]: replace with React.ReactElement[]

File: src/components/Checkbox/__tests__/Checkbox.spec.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (mukiienko) [reply src/components/Checkbox/__tests__/Checkbox.spec.tsx:3]: My point is that we already export `mount` to the global scope in `enzyme.setup.js`, so we don't need to export it again in the file. Let's use your suggestion if it solves the problem with the type and double exports.

File: src/components/RadioButton/RadioButtonGroup.tsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import React, { forwardRef } from 'react';
+import { bem } from '../../utils';
+import styles from './RadioButtonGroup.scss';
+
+export interface Props {
+    /** The name of the group this radio button belongs to */
+    name: string;
+    /** common onChange handler that will be passed to all children */
+    onChange?: () => void;
```
Comment (mukiienko) [inline src/components/RadioButton/RadioButtonGroup.tsx:9]: Is `onChange` optional? Does it have some arguments?

File: src/components/RadioButton/RadioButtonGroup.tsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    /** The radio buttons */
+    children: React.ReactElement[];
```
Comment (mukiienko) [inline src/components/RadioButton/RadioButtonGroup.tsx:11]: Should `children` only accept an array of React elements? Let's use one of the type from `types.d.ts`.

Comment (PavelKirvelTextkernel) [reply src/components/RadioButton/RadioButtonGroup.tsx:11]: From the original implementation it doesn't contain isRequired 
image

File: src/components/RadioButton/RadioButtonGroup.tsx (hunk 3 above)
Comment (PavelKirvelTextkernel) [reply src/components/RadioButton/RadioButtonGroup.tsx:9]: From the original implementation it doesn't contain isRequired
image

File: stories/molecules/RadioButton.tsx [hunk 5]
```diff
@@ -10,11 +10,12 @@ export default {
 export const _RadioButton = (args) => (
     <RadioButtonGroup name="my-group">
         <RadioButton {...args} />
-        <RadioButton value="option2" id="radio-2">
+        <RadioButton id="radio-2" {...{ value: 'option2' }}>
```
Comment (mukiienko) [inline stories/molecules/RadioButton.tsx:13]: Why did you rewrite option prop?

Comment (PavelKirvelTextkernel) [reply stories/molecules/RadioButton.tsx:13]: It shows error:
 Type '{ children: string; id: string; value: string; }' is not assignable to type 'IntrinsicAttributes & Props & RefAttributes<HTMLElement>'.   Property 'value' does not exist on type 'IntrinsicAttributes & Props & RefAttributes<HTMLElement>'.

As i understand value will pass like rest object <div {...block({ ...rest })}>

**Review by mukiienko — APPROVED** [review]

File: src/components/RadioButton/RadioButtonGroup.tsx [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+import { bem } from '../../utils';
+import styles from './RadioButtonGroup.scss';
+
+export interface Props {
+    /** The name of the group this radio button belongs to */
+    name: string;
+    /** common onChange handler that will be passed to all children */
+    onChange?: React.ChangeEventHandler<HTMLInputElement> | undefined;
+    /** The radio buttons */
+    children: null | undefined | React.ReactElement[];
```
[the remark this answers is not in the document]
Comment (PavelKirvelTextkernel) [reply src/components/RadioButton/RadioButtonGroup.tsx:11]: In cases when we get  response from BE and list will be empty

File: src/components/RadioButton/RadioButton.tsx [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    /** If the radio button should be disabled */
+    disabled?: boolean;
+    /** The label for the radio button */
+    children?: string;
+}
+
+const { block, elem } = bem('RadioButton', styles);
+
+export const RadioButton = forwardRef<HTMLElement, Props>(
+    ({ id, children = null, disabled = false, name = null, ...rest }, ref) => {
```
[the remark this answers is not in the document]
Comment (PavelKirvelTextkernel) [reply src/components/RadioButton/RadioButton.tsx:21]: I've taken default value from original implementation.

RadioButton.defaultProps = {
    name: null,
    disabled: false,
    children: null,
};

[the remark this answers is not in the document]
Comment (andrentk) [reply src/components/RadioButton/RadioButton.tsx:21]: I saw it, but do we really need to assign a default when the value is already undefined without doing anything?
IMO we can rely on typescript typing system to assume that or do you think it is better to leave it as it is?

File: src/components/RadioButton/RadioButtonGroup.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (andrentk) [reply src/components/RadioButton/RadioButtonGroup.tsx:11]: I think React.ReactElement already can be null (not 100% sure), maybe there is no need to have the other types there.

[the remark this answers is not in the document]
Comment (PavelKirvelTextkernel) [reply src/components/RadioButton/RadioButtonGroup.tsx:11]: I've checked just React.ReactElement[] and when i try to use null or undefined

File: src/components/RadioButton/RadioButton.tsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (PavelKirvelTextkernel) [reply src/components/RadioButton/RadioButton.tsx:21]: I've removed  default null for children and name

**Pull request merged**